client/src/elm/Pages/MessagingCenter/Utils.elm

```diff
@@ -108,11 +108,48 @@ surveyQuestionsAnswered surveyType surveyForm =
     Dict.size surveyForm == List.length surveyQuestions
 
 
-resolveSurveyScoreDialogState : ResilienceSurveyType -> Int -> SurveyScoreDialogState
-resolveSurveyScoreDialogState surveyType score =
+resolveSurveyScoreDialogState : NurseId -> ResilienceSurveyType -> Int -> ModelIndexedDb -> SurveyScoreDialogState
+resolveSurveyScoreDialogState nurseId surveyType score db =
```

Comment:
Note that this command is executed when new survey is saved.
After save is completed, new survey will appear at `db.resilienceSurveysByNurse`.
We can't be sure if it's there, when we display the dialog.
Therefore, lets filter out a survey, if it was created today (currentDate).